pkg/selector/nodevolumepath/selector.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package nodevolumepath
+
+import (
+	"context"
+
+	"go.uber.org/fx"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/controllers/config"
+	"github.com/chaos-mesh/chaos-mesh/pkg/selector/generic"
+	"github.com/chaos-mesh/chaos-mesh/pkg/selector/pod"
+)
+
+type SelectImpl struct {
+	c client.Client
+	r client.Reader
+
+	generic.Option
+}
+
+type NodeVolumePath struct {
+	nodeName string
+	// volumePath is a volumePath to the block device or directory on a node
+	volumePath string
+}
+
+func (n *NodeVolumePath) Id() string {
+	// The path may contain "/", but it doesn't matter
+	return n.nodeName + "/" + n.volumePath
+}
+
+func (impl *SelectImpl) Select(ctx context.Context, selector *v1alpha1.NodeVolumePathSelector) ([]*NodeVolumePath, error) {
```

Comment:
I feel like the nodevolumepath mod may probably be merged in the another PR, since it seems not complete.

Reply:
It's completed now (by returning some errors 😋 )